src/tests/profiler/unittest/releaseondetach.cs

```diff
@@ -58,19 +39,12 @@ public unsafe static int RunTest(string[] args)
             string rootPath = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
             string profilerPath = Path.Combine(rootPath, profilerName);
 
+            _profilerDone = new ManualResetEvent(false);
             Console.WriteLine($"Attaching profiler {profilerPath} to self.");
             ProfilerControlHelpers.AttachProfilerToSelf(ReleaseOnShutdownGuid, profilerPath);
 
-            // This warning is that the pointer to the volatile bool won't be treated as volatile,
-            // but that's ok. The loop aboive in WastTime is what needs to read it as volatile.
-            // The native part just sets it.
-            #pragma warning disable CS0420
-            fixed (bool *boolPtr = &_profilerDone)
-            {
-                PassBoolToProfiler(boolPtr);
-
-                WaitForDetach();
-            }
+            PassCallbackToProfiler(() => _profilerDone.Set());
+            _profilerDone.WaitOne();
```

Comment:
I like this a lot better. But we should probably have a timeout here.

Reply:
I'm open to adding the timeout, just want to know your reasoning. Are you concerned about the test getting in to an unknown state and taking forever for the testcase to time out?

Reply:
Yeah. The "this hung for some reason" case, just so it doesn't fireball all the tests that run along with this one. A ridiculous timeout is fine, say 5 min.